### PR TITLE
Single day demographics.

### PR DESCRIPTION
**Pull request by HassanJaveed84**
@mulby @brianhw please review.

### CODE REVIEW

File: edx/analytics/tasks/enrollments.py [hunk 1]
```diff
@@ -372,13 +372,17 @@ def required_table_tasks(self):
             ImportAuthUserProfileTask()
         )
 
+    @property
+    def query_date(self):
+        query_date = self.interval.date_b - datetime.timedelta(days=1)
```
Comment (HassanJaveed84) [inline edx/analytics/tasks/enrollments.py:378]: Is this what we want to do, filter by one day before interval end ?

Comment (mulby) [inline edx/analytics/tasks/enrollments.py:378]: Yup!

File: edx/analytics/tasks/enrollments.py [hunk 2]
```diff
@@ -391,7 +395,8 @@ def query(self):
                 ce.date,
                 ce.course_id,
                 IF(p.gender != '', p.gender, NULL)
-        """
+        """.format(date=self.query_date)
+        return query
```
Comment (mulby) [inline edx/analytics/tasks/enrollments.py]: We can probably revert all the changes for the gender task. Maybe add a comment explaining why we don't have a date filter here anymore?

File: edx/analytics/tasks/enrollments.py (hunk 1 above)
Comment (brianhw) [inline edx/analytics/tasks/enrollments.py:378]: Worth a comment?  (docstring?)

**Comment by brianhw** [comment]
Apart from the gender rollback, LGTM.  👍

**Comment by brianhw** [comment]
LGTM.   Squash and merge. 👍

**Comment by mulby** [comment]
:+1:

**Pull request merged**